cpu/cortexm_common/thread_arch.c

```diff
@@ -234,6 +238,15 @@ char *thread_arch_stack_init(thread_task_func_t task_func,
     return (char*) stk;
 }
 
+/* This function returns the number of bytes used on the ISR stack */
+int thread_arch_isr_stack_usage(void)
+{
+    uint32_t  *ptr = &_sstack;
+    while (*(ptr++) == STACK_CANARY_WORD) {
+    }
+    return ISR_STACKSIZE - (ptr - &_sstack);
```

Comment:
Since `ptr` and `&_sstack` are `uint32_t` pointers you will not get the number of bytes, but the number of `uint32_t`s. 
http://stackoverflow.com/questions/3238482/pointer-subtraction-confusion

I suggest two alternative solutions:
Either multiply `(ptr - &_sstack)` by `sizeof(*ptr)` or convert both pointers to `uint32_t` before the subtraction.